experimenter/evaluate_pipeline_new.py

```diff
@@ -0,0 +1,199 @@
+from experimenter.databases.d3m_mtl import D3MMtLDB
+
+
+def save_pipeline_run_to_d3m_db(pipeline_run_path: str):
+    """ 
+    Saves a pipeline run document to the d3m database.
+
+    Parameters
+    ----------
+    pipeline_run_path : path_like str
+        path to pipeline_run document
+
+    Returns:
+    ----------
+    TODO
+
+    Raises:
+    ----------
+    TODO
+    """
+    d3m_db = D3MMtLDB()
+    with open(pipeline_run_path) as pipeline_data:
+        pipeline_run = yaml.full_load(pipeline_data)
+    return D3MMtLDB().save_pipeline_run(pipeline_run)
+
+def evaluate_pipeline_on_problem(pipeline: str=None,
+    problem: str=None,
+    input: str=None,
+    random_seed: int=0,
+    data_pipeline_path: str=k_fold_split_path,
+    data_random_seed: int=0,
+    data_params=None,
+    scoring_pipeline: str=None,
+    scoring_params=None,
+    scoring_random_seed: int=0):
+    """ 
+    Evaluate pipeline on problem.
+    A less verbose form of running d3m's runtime cli 'evaluate' command.
+    See 'evaluate_pipeline_via_d3m_cli' for more options for running 
+    the 'evaluate' command.
+
+    Parameters
+    ----------
+    pipeline_path : path_like str
+        path to pipeline doc
+    problem_path : path_like str 
+        path to problem doc
+    input_path : path_like str
+        path to input full data
+    random_seed : int   
+        random seed to be used for pipeline run
+    data_pipeline_path: str
+        path to data prepation pipeline
+    data_random_seed: int
+        random_seed to be used in data preparation
+    data_params:
+        parameters for data preparation
+    scoring_params:
+        parameters for scoring pipeline
+    scoring_random_seed: int
+        random seed for scoring
+    scoring_pipeline: str
+        path to scoring pipeline
+
+    Returns:
+    ----------
+    None
+
+    Raises:
+    ---------------------------------
+    OSError
+        when a file cannot be opened
+    """
+    output_run_path = []
+    with open(pipeline, 'r') as data:
+        pipe = json.load(data)
+        output_run_path.append(pipe['id'])
+    with open(problem, 'r') as data:
+        prob = json.load(data)
+        output_run_path.append(prob['about']['problemID'])
+    output_run_path.append(str(random_seed))
+    #get the output run path
+    output_run_path = os.path.abspath(os.path.join('/data', 'Pipeline_Run', 
+                                                   '_'.join(output_run_path)+'.yaml'))
+    #create the directory
+    os.makedirs(os.path.dirname(output_run_path),exist_ok=True)
+    #evaluate pipeline
+    evaluate_pipeline_via_d3m_cli(pipeline=pipeline, problem=problem,
+        input=input, output_run=output_run_path,
+        random_seed=random_seed, data_pipeline_path=data_pipeline_path,
+        data_random_seed=data_random_seed, data_params=data_params, 
+        scoring_pipeline=scoring_pipeline, scoring_params=scoring_params,
+        scoring_random_seed=scoring_random_seed)
```

Comment:
`evaluate_pipeline_on_problem` is doing two things: creating the pipeline run path and calling `evaluate_pipeline_via_d3m_cli`. Let's get rid of `evaluate_pipeline_on_problem`, create a function for making the pipeline run path, and have `evaluate_pipeline_via_d3m_cli` call the function to create the pipeline run path.